tests/ui/manual_let_else_match.stderr

```diff
@@ -43,7 +43,7 @@ LL | /     let v = match f() {
 LL | |         Ok(v) => v,
 LL | |         Err(_) => return,
 LL | |     };
-   | |______^ help: consider writing: `let Ok(v) = f() else { return };`
+   | |______^ help: consider writing: `let Some(v) = f() else { return };`
```

Comment:
This suggestion gives the compile error because of mismatched types.

Reply:
Thank you for quick review, and I realized I've misunderstood the problem...
I'm sorry for bothering you, since this approach cannot handle cases like that so may I close the PR for now?

```rust
let v = match build_enum() {
  _ => continue,
  Variant::Bar(v) | Variant::Baz(v) => v,
};
```

Reply:
Okay, no worries! Anyway, thanks for your try to improve Clippy!

Reply:
Yeah pattern editing is tough. `rust-analyzer` has library functionality that can edit patterns, and it uses that library functionality for its "replace manual let else with `let-else`" tool. It's pretty cool.

For `|` patterns, the additional challenge is that they have to be surrounded by `(...)`.

Btw, if you want to fix this PR, you should remove "renamings of the bindings" from the list in the comments.